frontend/src/scenes/dashboard/dashboardLogic.tsx

```diff
@@ -1262,6 +1263,29 @@ export const dashboardLogic = kea<dashboardLogicType>([
                 actions.loadDashboard({ action: 'update' })
             }
         },
+        [dashboardsModel.actionTypes.updateDashboardSuccess]: ({ dashboard }) => {
+            if (values.dashboard == null || dashboard == null) {
+                return
+            }
+            if (values.dashboard.id !== dashboard.id) {
+                return
+            }
+            const newDashboard = { ...dashboard }
+            // Check if there are temporary filters
+            if (values.temporaryFilters.properties != null) {
```

Comment:
Not a fan of this nested if statements, but in this particular case i'm not sure if its worth extracting it out into another function.